reviewing/dependencies-hierarchy/src/getPkgInfo.ts

```diff
@@ -79,9 +80,16 @@ export function getPkgInfo (opts: GetPkgInfoOpts): PackageInfo {
   if (!version) {
     version = opts.ref
   }
-  const fullPackagePath = depPath
-    ? path.join(opts.virtualStoreDir ?? '.pnpm', depPathToFilename(depPath, opts.virtualStoreDirMaxLength), 'node_modules', name)
-    : path.join(opts.linkedPathBaseDir, opts.ref.slice(5))
+  let fullPackagePath = ''
+  if (depPath) {
+    if (opts.nodeLinker === 'hoisted') {
+      fullPackagePath = path.join(opts.linkedPathBaseDir, 'node_modules', name)
```

Comment:
so it will always give the location in the root node_modules? That isn't correct. Sometimes dependencies will be nested even with hoisted strategy. To get the correct location you need to take it from "hoistedLocations" in `node_modules/.modules.yaml`.